techmatters/hrm
# CHI-1115: Worker and status filtering support

Requires #166 and #167 to be merged first

Primary reviewer @murilovmachado 

## Description

* Combines the `GET /cases` and `POST cases/search` endpoints to use the same logic internally, only separate at the route layer
* Separates the notions of 'filters' (strict single value '=' matches, multiple value 'IN' matches or 'between' ranges for dates - all matched against a single source field) and 'search criteria' (more 'fuzzy' text based criteria, typically matched using 'ILIKE' comparisons and can be compared against multiple source fields)
* Reimplements the old 'closed cases' filter as a 'allow orphaned cases' filter and an 'excluded statuses' filter, which are set to 'false' and '['closed'] respectively when the 'closed cases' switch is set to 'off'
* counsellor, status and helpline filters support multiple values now

### Checklist
- [ X ] Corresponding issue has been opened
- [ X ] New tests added

### Related Issues
CHI-1115

### Verification steps

* Point a version of flex with filtering support at it and verify the filters return the expected cases
* Regression test search
